lib/src/builder_implementation.dart

```diff
@@ -4830,27 +5019,29 @@ _ClassDomain _createClassDomain(ClassElement type, _ReflectorDomain domain) {
 /// Answers true iff [element] can be imported into [generatedLibraryId].
 // TODO(sigurdm) implement: Make a test that tries to reflect on native/private
 // classes.
-bool _isImportable(
-    Element element, AssetId generatedLibraryId, Resolver resolver) {
-  return _isImportableLibrary(element.library, generatedLibraryId, resolver);
+Future<bool> _isImportable(
+    Element element, AssetId generatedLibraryId, Resolver resolver) async {
+  return await _isImportableLibrary(
+      element.library, generatedLibraryId, resolver);
 }
 
 /// Answers true iff [library] can be imported into [generatedLibraryId].
-bool _isImportableLibrary(
-    LibraryElement library, AssetId generatedLibraryId, Resolver resolver) {
-  Uri importUri = _getImportUri(library, generatedLibraryId);
+Future<bool> _isImportableLibrary(LibraryElement library,
```

Comment:
Is there any way we can cache the answer for this for all libraries up front?
It seems we would be able to save a lot of async work that way.

Reply:
Handled in #182.